istresearch/mailroom
# P4-3090 - Call details endpoint

# For the Reviewer

When this is complete, you should approve the PR via github.

# For the Reviewee

<Reminder PR Title should be JIRA_NUMBER and Useful Description>

## Summary
Fixed circleCI build. Added a new call details endpoint to allow reporting of non-IVR missed, incoming, and outgoing calls of a determined length for a given urn. 

#### Release Note
Added endpoint to allow non-IVR call reporting

#### Breaking Changes
None.

## Quality Assurance

You have gathered the following items:

Given all of the items above, you have updated your Application ORT at the following locations:
- **Features and Alerting**: <link to app ORT>Required.
- **P4 Alerting**: <link to p4-alerting PR>Required.

## Testing and Verification

1. Ensure you have a channel on the dev instance for testing. 
2. Using the dev instance, submit a POST request with the given payload for the endpoint, replacing the ID placeholder.
```
POST https://{{ DEV_INSTANCE_URL }}/mr/ivr/c/{{ CHANNEL_ID_HERE }}/call

{
    "urn": "+11234567890",
    "status": "in",
    "duration": 13
}
```
3. Check the calls tab in the UI under messages. The call will be listed as incoming for 13 seconds.
